cpp/src/arrow/engine/substrait/function_test.cc

```diff
@@ -601,6 +603,7 @@ std::shared_ptr<Table> GetOutputTableForAggregateCase(
 std::shared_ptr<acero::ExecPlan> PlanFromAggregateCase(
     const AggregateTestCase& test_case, std::shared_ptr<Table>* output_table,
     bool with_keys) {
+  static ExecContext ctx_for_ordered;
```

Comment:
The `static` makes the `ExecContext` instance global (and restricted to the function's scope), preventing its premature destruction. This is useful because the method it is passed to takes a pointer, rather than a `shared_ptr`, to an `ExecContext`. This instance is only used when the aggregation-case is ordered.

Reply:
This still feel strange to me that this is needed. Does something like this not work?
https://github.com/apache/arrow/blob/401ae19051e9120fedb8372a13cd68e08798c00a/cpp/src/arrow/acero/benchmark_util.cc#L49

Reply:
Looks like you can use this one?
https://github.com/apache/arrow/blob/main/cpp/src/arrow/acero/exec_plan.h#L72

Reply:
Here are [a couple of instances](https://github.com/search?q=repo%3Aapache%2Farrow%20%22static%20ExecContext%22&type=code) where this is used.

Reply:
Ok since there is existing usage of this pattern I am ok with it.